softicar/platform
# PLAT-1321: Suppress Concurrent Modification Exception in WorkflowAutoTransitionExecutionProgram

## Test Plan

First checkout the branch that is related to this pull request. Its name is "PLAT-1321", of course.

To reproduce the error checkout branch "main"!

To see that the error is fixed stay on branch "PLAT-1321".

Do `git cherry-pick 73d25f1e`. (`73d25f1e` is part of branch "PLAT-1321"!)

1. Start your TestServlet (`WebServiceTestService`).
2. Log-in as "admin.user".
3. Go to page "Daemon Watchdog State" and start the "watchdog".
4. Go to page "Workflow Demo Objects"  and start the workflow of the existing record.
5. _**Immediately**_ after (4) go to page "My Tasks".
6. If you don't see a task, reload the page "My Tasks" each second.
7. If you see a task, _**immediately**_ approve or reject it.
8. If you are too late, you will see the message "This action is not available anymore." Then shut down your TestServlet and restart at (1).

If you don't see the "too late message", go to page "Program Executions". If you are on branch "main", you will see that one execution of `WorkflowAutoTransitionExecutionProgram` has the error we want to get rid of. If you are on branch "PLAT-1321", there will be no error anymore.

### With error:
![Screenshot_2023-03-08_14-56-23](https://user-images.githubusercontent.com/74466399/223731841-ab7abe67-7911-4053-9773-9aba18532521.png)

### Without error:

![PLAT-1321](https://user-images.githubusercontent.com/74466399/224331449-d12e7b9d-7fd3-47f5-9463-7c967fe84a92.png)
